### PR TITLE
TB - 79 Update documentation for kafka and authentication changes

### PR DESCRIPTION
**Pull request by arran-standish**
With the recent changes to introducing Kafka support and changes to authentication, so this is to keep the documentation in sync.

### CODE REVIEW

File: docs/api/introduction/authentication.md [hunk 1]
```diff
@@ -11,14 +11,143 @@ description: Authentication for the OpenHIM API calls
...
+
+This is very similar to the Local authentication, except that it uses the Basic passport module and every request to any endpoint should be authenticated.
+
+For more information on basic authentication in Passport.js, check out: https://www.passportjs.org/packages/passport-http/
+
+If this type of authentication is enabled then you can add this header to your request: 
+
+```json
+"headers": { 
+    // ...,
```
Comment (bradsawadye) [inline docs/api/introduction/authentication.md:81]: Please remove this

Comment (nour-borgi) [reply docs/api/introduction/authentication.md]: cool done :)

File: docs/dev-guide/design-details.md [hunk 2]
```diff
@@ -141,7 +141,9 @@ In addition, the ability to store orchestration steps exists in the structure. W
...
+    
+> **Note:** By default all routes defined in a channel are run simultaneously along side the primary route. However it is possible to alter this behavior by setting a route to wait for the primary route. In which case the primary route and any non-waiting non-primary routes are run simultaneously, while all waiting non-primary routes will only run once the primary route has finished. It is possible to futher alter this process by specifying specific status codes the primary route should respond with before running the waiting non-primary routes. This enables, for example, the ability to send a request to a logging service on the event that the primary route returns a 401 or 500.
```
Comment (bradsawadye) [inline docs/dev-guide/design-details.md:146]:
```suggestion
> **Note:** By default all routes defined in a channel are run simultaneously along side the primary route. However it is possible to alter this behavior by setting a route to wait for the primary route. In which case the primary route and any non-waiting non-primary routes are run simultaneously, while all waiting non-primary routes will only run once the primary route has finished. It is possible to further alter this process by specifying specific status codes the primary route should respond with before running the waiting non-primary routes. This enables, for example, the ability to send a request to a logging service on the event that the primary route returns a 401 or 500.  
```

File: docs/how-to/how-to-login-using-keycloak.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+### Notes
+
+Clicking on Sign-in with keycloak will redirect you to Keycloak Login UI.
+
+<img alt="Login With keycloak UI" src={useBaseUrl('img/keycloak-login-ui.png')} />
+
+After entering the Keycloak User credentials and submiting you will be redirected to the dashboard.
+
+Make sure that Keycloak is on and running to be able to authenticate successfully and that you configured a user with group permissions and details that you need.
```
Comment (bradsawadye) [inline docs/how-to/how-to-login-using-keycloak.md:65]:
```suggestion
Make sure that Keycloak is on and running to be able to authenticate successfully and that you configured a user with group permissions and details that you need.
```
```suggestion
Make sure that Keycloak is on and running to be able to authenticate successfully and that you configured a user with group permissions and details that you need.
```
```suggestion
Make sure that Keycloak is on and running to be able to authenticate successfully and that you configured a user with group permissions and details that you need.
```
```suggestion
Keycloak should be running and configured with a user that has group permissions and details that will be needed.
```

File: docs/user-guide/adding-users.md [hunk 4]
```diff
@@ -88,3 +88,13 @@ The statuses are as follows:
...
 
 If you find a field that is not described here, please let us know by [filing an issue on github](https://github.com/jembi/openhim-core-js/issues/new) with the 'documentation' label.
+
+## Keycloak Single Sign-On (SSO)
+
+Openhim console and core are supporting SSO authentication (Keycloak identity manager).
+Enabling SSO in the openhim console and core will allow:
+
+1. Authentication: A user can be logged in using keycloak credentials.
+2. Profile setup: The user authenticated using Keycloak will have his profile filled automatically with info provided by Keycloak.
```
Comment (bradsawadye) [inline docs/user-guide/adding-users.md:98]:
```suggestion
2. Profile setup: The user authenticated using Keycloak will have a profile created automatically with info provided by Keycloak. 
```

**Pull request merged**